src/vendor/utils.js

```diff
@@ -212,6 +212,35 @@ function concatField(...many) {
   return output;
 }
 
+/*
+Represent JSON object: map from string keys to values
+ */
+class Data {}
+
+const OBJECT_CONSTRUCTOR = {}.constructor;
+
+/*
+ * Check if the `val` is Data
+ * Direct instances of Object are also considered data
+ */
+function isData(val) {
+  if (missing(val)) {
+    return false;
+  }
+
+  if (val[Symbol.iterator]) {
+    return false;
+  }
+
+  return val.constructor === OBJECT_CONSTRUCTOR || val instanceof Data;
+}
+
+function sleep(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
+}
```

Comment:
A couple of questions re: `isData` and `sleep`
* what is `isData` checking for?
* why do we need to check the constructor property?
* why `sleep`?

Reply:
@helfi92 

`isData` is checking for `Data` objects: 

> Represent JSON object: map from string keys to values

Plain objects, objects created with `{}`, can act like `Data`.  So they are included.

`sleep()` is used to insert delays into an asynch function. `sleep(0)` is intended to let other promise chains execute.